cmake/common.cmake

```diff
@@ -284,6 +294,7 @@ endfunction()
 function(use_client_mbedtls name)
   target_include_directories(${name} PRIVATE ${CLIENT_MBEDTLS_INCLUDE_DIR})
   target_link_libraries(${name} PRIVATE ${CLIENT_MBEDTLS_LIBRARIES})
+  target_compile_definitions(${name} PRIVATE -DMOD_MBEDTLS)
```

Comment:
We should summarise what the `MOD_MBEDTLS` (enables curve25519 support) somewhere, probably near the definition here.

Reply:
But we can't really use it against the current server, so I think we just need to remove this as well.

For reference, the PR merging Christoph's x25519 support upstream can be found here: https://github.com/ARMmbed/mbed-crypto/pull/140

Reply:
Switched client side over to vanilla mbedtls too.